editor/src/clj/editor/editor_extensions/ui_components.clj

```diff
@@ -218,50 +215,40 @@
-                                           (let [ret {:fx/type lifecycle}]
-                                             (if maybe-column
-                                               (let [{:keys [grow]} maybe-column]
-                                                 (cond-> ret grow (assoc grow-key :always)))
-                                               ret)))
-                                         constraints)))
-
-(def ^:private ^{:arglists '([props padding])} apply-grid-spacing
-  (let [spacing->style-class (coll/pair-map-by identity #(str "ext-grid-spacing-" (name %)) (:spacing ui-docs/enums))]
-    (fn apply-grid-spacing [props spacing]
-      {:pre [(some? spacing)]}
-      (if-let [style-class (spacing->style-class spacing)]
-        (fxui/add-style-classes props style-class)
-        (if (number? spacing)
-          (assoc props :hgap spacing :vgap spacing)
-          (throw (AssertionError. (str "Invalid spacing: " spacing))))))))
+(defn- apply-constraints [props props-key lifecycle grow-key constraints]
+  (assoc props props-key (mapv (fn [maybe-column]
+                                 (let [ret {:fx/type lifecycle}]
+                                   (if maybe-column
+                                     (let [{:keys [grow]} maybe-column]
+                                       (cond-> ret grow (assoc grow-key :always)))
+                                     ret)))
+                               constraints)))
 
 (defn- grid-view [{:keys [children rows columns alignment padding spacing]
                    :or {spacing :medium}}]
-  (-> {:fx/type fx.grid-pane/lifecycle}
-      (apply-grid-spacing spacing)
-      (cond->
-        children (assoc :children
-                        (into []
-                              (coll/mapcat-indexed
-                                (fn [row row-children]
-                                  (eduction
-                                    (keep-indexed
-                                      (fn [column child]
-                                        (when child
-                                          (let [{:keys [row_span column_span]
-                                                 :or {row_span 1 column_span 1}} (:props (meta child))]
-                                            (assoc child :grid-pane/row row
-                                                         :grid-pane/column column
-                                                         :grid-pane/row-span row_span
-                                                         :grid-pane/column-span column_span)))))
-                                    row-children)))
-                              children))
-        rows (apply-constraints fx.row-constraints/lifecycle :vgrow rows)
-        columns (apply-constraints fx.column-constraints/lifecycle :hgrow columns)
-        padding (apply-padding padding)
-        alignment (apply-alignment alignment))))
+  (cond->
+    {:fx/type fxui/grid
+     :spacing spacing}
```

Comment:
I started working on unifying the views between editor scripts and clojure editor code, one bit at a time...